cpp/velox/memory/VeloxMemoryManager.cc

```diff
@@ -62,26 +63,28 @@ class ListenableArbitrator : public velox::memory::MemoryArbitrator {
   }
 
   void reserveMemory(velox::memory::MemoryPool* pool, uint64_t) override {
-    growPool(pool, memoryPoolInitCapacity_);
+    std::lock_guard<std::recursive_mutex> l(mutex_);
```

Comment:
👍 
could we remove AllocationListener's mutex?

Reply:
Let's keep that mutex for now since this one only works in Velox backend.